### PR TITLE
Get Arguments Directly From Pointers

### PR DESCRIPTION
**Pull request by vixalien**
> I lacked a better title for this PR :^(

This PR is significant because it implements passing around arguments in their raw pointer form instead of allocating new ArrayBuffers.

This is done by creating a `BigUint64Array`, passed as an argument to FFI functions (`gi_(function|method|callback)_invoke`). The FFI functions directly manipulate and set the pointers as indices of the `BigUint64Array`, and we can get the assigned pointers directly from the BigUint64Array. Here's some pseudocode to explain this:

```js
const array = BigUint64Array(outArgDetails.length);

// the array is initially empty
console.log(array) // [0,0,0,0,....]

// invoke the function with the array as outArgs
g.function_info.invoke(
  function_name,
  //...
  outArgs,
  // ....
);

// now the array has been modified to have pointers
console.log(array) // [28933n, 329839n, 23893495834n, ...]

// now we can use the pointers directly (in the case of numbers, the values are not pointers but the numbers themselves directly
```

Thanks for taking a look at this.

I also created a new file, `examples/bytes.ts`, a temporary file to test things out since deno_gi doesn't have actual tests at the moment (I'm working on it...). You can delete this before merging. Thanks!

### CODE REVIEW

File: src/types/callable.js [hunk 1]
```diff
@@ -62,14 +69,12 @@ export function parseCallableArgs(info) {
   };
 
   const initOutArgs = () => {
-    const outArgs = outArgsDetail.map((d) => initArgument(d.type));
-
-    return outArgs;
+    return new BigUint64Array(outArgsDetail.map((d) => initArgument(d.type)));
```
Comment (vixalien) [inline src/types/callable.js:72]: This isn't quite correct. I think `initArgument` should return a pointer to a memory of the size of the interface (if given an interface) and not initialise a new object. Please advise!

File: src/types/callable/function.js [hunk 2]
```diff
@@ -12,14 +12,14 @@ export function createFunction(info) {
...
     const outArgs = initOutArgs();
 
-    const error = new ArrayBuffer(16);
-    const returnValue = new ArrayBuffer(8);
+    const error = new ArrayBuffer(8);
+    const returnValue = new BigUint64Array(1);
 
     const success = g.function_info.invoke(
       info,
       new BigUint64Array(inArgs),
```
Comment (vixalien) [inline src/types/callable/function.js:20]: maybe `parseInArgs` should also return a `BigUint64Array` for consistency?

**Comment by vixalien** [comment]
Maybe it's time to review this next?

File: examples/bytes.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { require } from "../mod.ts";
```
Comment (vixalien) [inline examples/bytes.ts:1]: This needs to be removed before the PR can be merged

File: src/types/argument.js [hunk 4]
```diff
@@ -24,58 +24,48 @@ export function initArgument(type) {
...
-
     case GITypeTag.INT64:
-      return dataView.getBigInt64();
-
     case GITypeTag.FLOAT:
-      return dataView.getFloat32();
-
-    case GITypeTag.DOUBLE:
-      return dataView.getFloat64();
+    case GITypeTag.DOUBLE: {
```
Comment (vixalien) [inline src/types/argument.js:60]: Not sure if this is the best idea. Maybe it's better to always return BigInts for UINT64 and INT64 **ONLY**

**Comment by ahgilak** [comment]
Sorry it's taking so long, I'm a bit busy these days. I'll review this by the end of the week.

**Comment by vixalien** [comment]
> Sorry it's taking so long, I'm a bit busy these days. I'll review this by the end of the week.

Hey! No worries. Take your time, and you don't have to review these ASAP. I just create a PR for every thing I change so the changes don't get lost. Please don't feel pressured to review this and take your time!

File: src/types/callable/function.js (hunk 2 above)
Comment (ahgilak) [reply src/types/callable/function.js:20]: in instance methods, you need to add self argument later, so it's easier to use dynamic array.

**Comment by vixalien** [comment]
I've rebased the code and removed the `bytes.ts` file. I plan to reintroduce plausible tests later.

File: src/types/callable.js (hunk 1 above)
Comment (ahgilak) [reply src/types/callable.js:72]: I'm not entirely sure about that. If you managed to do it the better way, introduce it in a new PR.

**Pull request merged**